erasure-coding/src/lib.rs

```diff
@@ -228,19 +200,19 @@ pub fn reconstruct<'a, I: 'a>(n_validators: usize, chunks: I)
 
 /// An iterator that yields merkle branches and chunk data for all chunks to
 /// be sent to other validators.
-pub struct Branches<'a> {
+pub struct Branches<'a, I: AsRef<[u8]>> {
```

Comment:
should just be `I` on the struct. it's better to put bounds only on implementations.